backend/municipal-services/ingestion-service/app/api/endpoints/template_generation.py

```
@@ -0,0 +1,112 @@
+    mdms_client = MDMSClient(mdms_url)
+    try:
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        output_filename = f"facility_ingestion_template_{timestamp}.xlsx"
+        output_file_path = create_temp_file(suffix=".xlsx")
+        try:
+            facility_schema = mdms_client.fetch_facility_schema(request_info=request_info)
+            boundary_data = facility_service.get_all_boundaries(request_info)
+        except Exception as e:
+            logger.error(f"Error fetching data from external services: {e}")
+            cleanup_temp_file(output_file_path)
+            raise HTTPException(status_code=502, detail=f"External service error: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)